packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -431,11 +477,19 @@ export class TransformerResolver implements TransformerResolverProvider {
       `
       : '$util.toJson($ctx.prev.result)';
 
+    const initResolverMappingTemplate: FunctionRuntimeTemplate = isJsResolverFnRuntime(this.runtime)
```

Comment:
This function implementation could benefit from splitting things out in a cleaner way. A part of me feels bad for tacking on more runtime specific conditional checks here. But a bigger part wanted to keep the scope of this (already large) PR to a minimum.